CloudEvent 'data' attributes are passed to function inputs as strings

## Bug Report
While trying to subscribe to events with a python or java function if the event includes a data payload both functions fail on deserializing their input. This can be seen best with these two executions:

```
{
    "blocking": true,
    "executedTime": 1526513318,
    "faasId": "a5666367-ade7-4b79-aa39-569770fe5fdd",
    "finishedTime": 1526513318,
    "functionId": "a7a43694-9916-4b59-ab41-231f24db9b5b",
    "functionName": "java-fn",
    "input": {
        "name": "Jon",
        "place": "Winterfell"
    },
    "logs": {
        "stderr": null,
        "stdout": null
    },
    "name": "2a5ae81a-5588-44e6-b9f1-ec0cbae7f9cf",
    "output": "Hello, Jon from Winterfell",
    "reason": null,
    "secrets": [],
    "services": null,
    "status": "READY",
    "
```

and

```
{
    "event": {
        "cloud-events-version": "0.1",
        "content-type": "application/json",
        "event-id": "6f3fee2b-8ffb-49cb-b1ea-aa6601c1fada",
        "event-time": "2018-05-16T16:26:46.400-07:00",
        "event-type": "symphony2",
        "namespace": "dispatchframework.io",
        "source-id": "dispatch",
        "source-type": "dispatch"
    },
    "executedTime": -62167134794,
    "faasId": "a5666367-ade7-4b79-aa39-569770fe5fdd",
    "finishedTime": 1526513206,
    "functionId": "a7a43694-9916-4b59-ab41-231f24db9b5b",
    "functionName": "java-fn",
    "input": "{\"name\" : \"Jon\", \"place\" : \"Winterfell\"}",
    "logs": {
        "stderr": [
            "com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $",
            "\tat com.google.gson.Gson.fromJson(Gson.java:900)",
            "\tat com.google.gson.Gson.fromJson(Gson.java:953)",
            "\tat io.dispatchframework.javabaseimage.SimpleFunctionExecutor.applyFunction(SimpleFunctionExecutor.java:90)",
            "\tat io.dispatchframework.javabaseimage.SimpleFunctionExecutor.execute(SimpleFunctionExecutor.java:61)",
            "\tat io.dispatchframework.javabaseimage.POJFunctionServer$ExecFunction.lambda$handleRequest$0(POJFunctionServer.java:73)",
            "\tat io.undertow.io.AsyncReceiverImpl.receiveFullString(AsyncReceiverImpl.java:140)",
            "\tat io.undertow.io.AsyncReceiverImpl.receiveFullString(AsyncReceiverImpl.java:79)",
            "\tat io.dispatchframework.javabaseimage.POJFunctionServer$ExecFunction.handleRequest(POJFunctionServer.java:72)",
            "\tat io.undertow.server.handlers.PathHandler.handleRequest(PathHandler.java:94)",
            "\tat io.undertow.server.Connectors.executeRootHandler(Connectors.java:360)",
            "\tat io.undertow.server.protocol.http.HttpReadListener.handleEventWithNoRunningRequest(HttpReadListener.java:255)",
            "\tat io.undertow.server.protocol.http.HttpReadListener.handleEvent(HttpReadListener.java:136)",
            "\tat io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:151)",
            "\tat io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:92)",
            "\tat io.undertow.server.protocol.http.HttpOpenListener.handleEvent(HttpOpenListener.java:51)",
            "\tat org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)",
            "\tat org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:291)",
            "\tat org.xnio.ChannelListeners$10.handleEvent(ChannelListeners.java:286)",
            "\tat org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)",
            "\tat org.xnio.nio.QueuedNioTcpServer$1.run(QueuedNioTcpServer.java:129)",
            "\tat org.xnio.nio.WorkerThread.safeRun(WorkerThread.java:582)",
            "\tat org.xnio.nio.WorkerThread.run(WorkerThread.java:466)",
            "Caused by: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at path $",
            "\tat com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:162)",
            "\tat com.google.gson.internal.bind.JsonTreeReader.beginObject(JsonTreeReader.java:87)",
            "\tat com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:183)",
            "\tat com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145)",
            "\tat com.google.gson.Gson.fromJson(Gson.java:888)",
            "\t... 21 more"
        ],
        "stdout": null
    },
    "name": "401729bb-7d10-406f-9b1a-09156aef5148",
    "reason": null,
    "secrets": null,
    "services": null,
    "status": "READY",
    "tags": []
}
```

### Expected behavior
<!--- What did you expect to happen --->
Both functions should execute succesfully

### Current behavior
<!--- what actually happened --->
When function is invoked from an event the function fails.

### Steps to reproduce
<!--- How can we reproduce the issue --->

### Impact
<!--- How much does this issue impact you or your business --->
- [ ] Low - Annoyance, but does not impact business or functionality
- [ ] Medium - Issue can be worked around, but is causing pain
- [x] High - Blocker

### Possible solution
<!--- Only if you have an idea how to fix it --->
When a subscription invokes a function it should pass the data payload as the correct type as based on the content-type.

### Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Dispatch CLI version (or git commit):
* Dispatch Chart version (or image[s] tag):
* Operating System and version:
* Kubernetes version (and distribution):
* Etc (any other useful environmental information):

--- a/pkg/event-manager/subscriptions/manager.go
+++ b/pkg/event-manager/subscriptions/manager.go
@@ -7,6 +7,7 @@ package subscriptions
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"sync"
 
@@ -136,10 +137,15 @@ func (m *defaultManager) handler(sub *entities.Subscription) func(context.Contex
 func (m *defaultManager) runFunction(fnName string, event *events.CloudEvent, secrets []string) {
 	defer trace.Tracef("function:%s", fnName)()
 
+	processedData, err := m.processEventData(event)
+	if err != nil {
+		log.Warnf("Unable to process event payload: %+v", err)
+	}
+
 	run := v1.Run{
 		Blocking:     false,
 		FunctionName: fnName,
-		Input:        event.Data,
+		Input:        processedData,
 	}
 	eventCopy := *event
 	eventCopy.Data = ""
@@ -152,4 +158,19 @@ func (m *defaultManager) runFunction(fnName string, event *events.CloudEvent, se
 	}
 	log.Debugf("Function %s returned %+v", result.FunctionName, result.Output)
 	return
+}
+
+func (m *defaultManager) processEventData(event *events.CloudEvent) (interface{}, error) {
+	switch event.ContentType {
+	case "application/json":
+		var jsonData interface{}
+		err := json.Unmarshal([]byte(event.Data), &jsonData)
+		if err != nil {
+			return nil, errors.Wrap(err, "error when unmarshaling JSON data")
+		}
+		return jsonData, nil
+	default:
+		// for every other content type we pass data as is
+		return event.Data, nil
+	}
 }

